### PR TITLE
Release/11.1.0

### LINKED ISSUE #247: implement ability to configure global middleware
it should run before all other configured component middleware regardless of web route accessed

## use case

add 'same-origin' response header to every response from server without having to add a middleware function for each web route

### PR DESCRIPTION
**Pull request by southbite**
fixes #247

### CODE REVIEW

**Comment by JensEggers** [comment]
please reference issues to be closed by this PR in its Description field

File: lib/mesh.js [hunk 1]
```diff
@@ -975,6 +980,63 @@ Mesh.prototype._initializeRootWebRoutes = function(callback) {
...
+    return function(req, res, next) {
+      if (!_this.exchange[middleware.component]) {
+        return next(new Error(`Component ${middleware.component} not found for global middleware`));
+      }
+      if (!_this.exchange[middleware.component][methodName]) {
+        return next(
+          new Error(`Method ${middleware.component}.${methodName} not found for global middleware`)
+        );
+      }
+      _this.exchange[middleware.component][methodName].call(
```
Comment (JensEggers) [inline lib/mesh.js]: is _this.exchange[middleware.component][methodName] not already bound to _this.exchange[middleware.component]?

File: lib/mesh.js [hunk 2]
```diff
@@ -975,6 +980,63 @@ Mesh.prototype._initializeRootWebRoutes = function(callback) {
...
+        _this.exchange[middleware.component],
+        req,
+        res,
+        next
+      );
+    };
+  });
+  //error handler for failed middleware
+  //eslint-disable-next-line
+  methods.push(function(err, req, res, next) {
```
Comment (JensEggers) [inline lib/mesh.js]: question on connect - does it treat a middleware differently if it has the (err, req, res, next) signature?

Comment (JensEggers) [reply lib/mesh.js]: i.e. wondering what stops the error handler from being invoked under normal circumstances

Comment (southbite) [reply lib/mesh.js]: Based on its arguments,it is not invoked, see: https://github.com/senchalabs/connect#error-middleware

File: lib/mesh.js (hunk 1 above)
Comment (southbite) [reply lib/mesh.js]: I have made it null, just did a quick test to ensure that the "this" was correct (which it was)

**Pull request merged**